components/camel-box/camel-box-api/src/main/java/org/apache/camel/component/box/api/BoxFilesManager.java

```diff
@@ -162,7 +162,7 @@ public BoxFile uploadFile(String parentFolderId, InputStream content, String fil
             }
             BoxFile boxFile = null;
             boolean uploadNewFile = true;
-            if (check) {
+            if (check != null && check) {
```

Comment:
ObjectHelper would be better? :-)

Reply:
There is toBoolean method, but it return null when parameter is null, hence the validation is redundant